compiler/rustc_hir_typeck/src/method/probe.rs

```diff
@@ -950,29 +950,17 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
         let trait_substs = self.fresh_item_substs(trait_def_id);
         let trait_ref = self.tcx.mk_trait_ref(trait_def_id, trait_substs);
 
-        if self.tcx.is_trait_alias(trait_def_id) {
-            // For trait aliases, assume all supertraits are relevant.
-            let bounds = iter::once(ty::Binder::dummy(trait_ref));
-            self.elaborate_bounds(bounds, |this, new_trait_ref, item| {
-                let new_trait_ref = this.erase_late_bound_regions(new_trait_ref);
+        for expansion in traits::expand_trait_aliases(
```

Comment:
Can you not merge these two branches? Really, the only thing you need to do here is replace the `elaborate_bounds` call. It's a bit redundant, but I'd rather keep trait alias code as far away from regular code as possible.

Reply:
Sure, I'll put the branch back.

However, `elaborate_bounds` loops over the trait's items, either calling `record_static_candidate` or invoking the callback as appropriate; so we'll either need to insert that explicit loop into this method (as has already been done for the non-alias case) or else factor that loop out into some `expand_aliases` analogue of the `elaborate_bounds` method.  Indeed the same looping logic actually appears in quite a few places in the file, so factoring it into a helper method might be wise anyway (for another PR).

Reply:
Feel free to do some factoring if you'd like in this file.